src/backend/distributed/commands/function.c

```diff
@@ -187,22 +187,62 @@ create_distributed_function(PG_FUNCTION_ARGS)
 	EnsureFunctionOwner(funcOid);
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
-	ErrorIfFunctionDependsOnExtension(&functionAddress);
 
 	/*
-	 * when we allow propagation within a transaction block we should make sure to only
-	 * allow this in sequential mode
+	 * If the function is owned by an extension, only update the
+	 * pg_dist_object, and not propagate the CREATE FUNCTION. Function
+	 * will be created by the virtue of the extension creation.
 	 */
-	EnsureSequentialModeForFunctionDDL();
+	if (IsObjectAddressOwnedByExtension(&functionAddress, &extensionAddress))
+	{
+		if (IsObjectDistributed(&functionAddress))
+		{
+			ereport(ERROR, (errmsg("Function(%s) is already distributed.",
+								   get_func_name(funcOid))));
+		}
+
+		/*
+		 * Ensure corresponding extension is in pg_dist_object.
+		 */
+
+		Oid citusExtensionOid = get_extension_oid("citus", false);
+
+		if (citusExtensionOid == extensionAddress.objectId)
+		{
+			/* Citus extension is a special case */
```

Comment:
comment would have more value explaining _why_ citus is a special case.

My gut feeling suggests it is because we don't want citus functions to be pushed down to workers, whatever the user says. The comment in its current form is not super descriptive.

Reply:
Added couple of more lines, more than happy to be more descriptive, please let me know.